msp/mspimpl.go

```diff
@@ -890,27 +890,40 @@ func (msp *bccspmsp) getCertificationChainIdentifierFromChain(chain []*x509.Cert
 // is regenerated to have a Low-S signature.
 func (msp *bccspmsp) sanitizeCert(cert *x509.Certificate) (*x509.Certificate, error) {
 	if isECDSASignedCert(cert) {
-		// Lookup for a parent certificate to perform the sanitization
-		var parentCert *x509.Certificate
-		chain, err := msp.getUniqueValidationChain(cert, msp.getValidityOptsForCert(cert))
-		if err != nil {
-			return nil, err
-		}
+		// We sanitize the root CA certificate without an attempt to
+		// build the chain, for two reasons:
+		// 1. It's a root CA cert hence no need to actually build the chain
+		// 2. Second, since v1.19 golang introduced an optimization on how
+		// certificates are compared, while the previous code was worked under
+		// the assumption that due to difference in the bitwise comparision
+		// of two root CA certificates one before and one after the sanitization
+		// it will still be able to introduce into chain self signed cert. However,
+		// after the optimization v1.19 both cert with satinizied and without
+		// sanitized certiticates considered equal and therefore the code
+		// misses to build the chain.
+		// Hence assiggning parent cert to be the the cert itself, and if
+		// this is not the case we take parent from the certs chain.
+		parentCert := cert
+		if cert.CheckSignatureFrom(cert) != nil || !cert.IsCA {
+			// Lookup for a parent certificate to perform the sanitization
+			validityOpts := msp.getValidityOptsForCert(cert)
+			chain, err := msp.getUniqueValidationChain(cert, validityOpts)
+			if err != nil {
+				return nil, err
+			}
 
-		// at this point, cert might be a root CA certificate
-		// or an intermediate CA certificate
-		if cert.IsCA && len(chain) == 1 {
-			// cert is a root CA certificate
-			parentCert = cert
-		} else {
-			parentCert = chain[1]
+			// at this point, cert might be a root CA certificate
+			// or an intermediate CA certificate
+			if cert.IsCA && len(chain) == 1 {
+				// cert is a root CA certificate
+				parentCert = cert
```

Comment:
sorry, I'm still a little puzzled by this: if the length of the chain is 1, it's a root CA - at which point iiuc we would not be in this block anyway. What am I missing?

(and in case I'm wrong, we should add some tests because we don't seem to test this if case